solidity/src/Polymer7683.sol

```diff
@@ -0,0 +1,175 @@
+    ICrossL2ProverV2 public immutable prover;
+    uint256 public immutable localChainId;
+    mapping(uint256 => address) public destinationContracts;
+
+
+    // ============ Events ============
+    /**
+     * @notice Event emitted when a destination contract is updated
+     * @param chainId The chain ID for the destination
+     * @param contractAddress The new contract address
+     */
+    event DestinationContractUpdated(uint256 indexed chainId, address contractAddress);
+
+    // ============ Errors ============
+    error InvalidProof();
+    error InvalidChainId();
+    error InvalidEmitter();
+    error InvalidEventData();
+    error InvalidDestinationContract();
+    error UnregisteredDestinationChain();
+
+    // ============ Constructor ============
+    /**
+     * @notice Initializes the Polymer7683 contract with the specified Prover and PERMIT2 address.
+     * @param _prover The address of the Polymer CrossL2Prover contract
+     * @param _permit2 The address of the permit2 contract
+     * @param _localChainId The chain ID of the chain this contract is deployed on
+     */
+    constructor(
+        ICrossL2ProverV2 _prover,
+        address _permit2,
+        uint256 _localChainId
+    ) BasicSwap7683(_permit2) {
+        prover = _prover;
+        localChainId = _localChainId;
+    }
+
+    // ============ Admin Functions ============
+    function setDestinationContract(uint256 chainId, address contractAddress) external onlyOwner {
+        if (contractAddress == address(0)) revert InvalidDestinationContract();
+        destinationContracts[chainId] = contractAddress;
+        emit DestinationContractUpdated(chainId, contractAddress);
+    }
+
+    // ============ External Functions ============
+    /**
+     * @notice Process a settlement proof from a destination chain
+     * @param eventProof The proof of the Fill event from the destination chain
+     * @param eventProof The proof of the Fill event from the destination chain
+     */
+    function handleSettlementWithProof(bytes calldata eventProof) external {
+        // Verify event using Polymer prover
+        (
+            uint32 provenChainId,
+            address emitter,
+            ,  // topics
+            bytes memory data
+        ) = prover.validateEvent(eventProof);
+
+        // Verify destination contract is registered for the proven chain
+        address expectedEmitter = destinationContracts[provenChainId];
+        if (expectedEmitter == address(0)) revert UnregisteredDestinationChain();
+
+        // Validate emitter matches registered destination
+        if (emitter != expectedEmitter) revert InvalidEmitter();
+
+        // Decode data from the Filled event format
+        (bytes32 eventOrderId, bytes memory originData, bytes memory fillerData) = abi.decode(data, (bytes32, bytes, bytes));
+
+        // Process settlement with the message origin, sender, order ID, and receiver
+        _handleSettleOrder(
```

Comment:
The `_handleSettleOrder` doesn't revert when trying to settle a wrong order to allow processing a batch of settlements without reverting the whole batch because of a single invalid order, the same for the `_handleRefundOrder`

In your case, you are handling single orders in both actions so I wonder if it isn't a good idea reverting when the conditions are not met.  WDYT?

Reply:
yeah, totally make sense. i added a check at the end and revert if the status hasn't been updated.